### PR TITLE
UI: Fixes security vulnerability in npm dependencies

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes

- Bumps react scripts to latest version and updates
  snapshot of Readme component
- Bumps node and npm to latest version

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "react-hotkeys-hook": "^3.3.1",
     "react-markdown": "^5.0.3",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "^4.0.1",
+    "react-scripts": "4.0.3",
```
Comment (PuneetPunamiya) [inline ui/package.json:32]: Is this the only dependency we need to update ??

File: ui/src/components/Readme/__snapshots__/Readme.test.tsx.snap [hunk 2]
```diff
@@ -23,10 +23,7 @@ exports[`should render the readme file 1`] = `
             #
           </span>
           <span className=\\"token title\\" style={{...}}>
-             
-          </span>
-          <span className=\\"token code\\" style={{...}}>
-            &#96;az&#96;
+             &#96;az&#96;
```
Comment (PuneetPunamiya) [inline ui/src/components/Readme/__snapshots__/Readme.test.tsx.snap:26]: Does the above dependency change the snapshot ??

File: ui/package.json (hunk 1 above)
Comment (PuneetPunamiya) [reply ui/package.json:32]: Should we also check if there are updates for the other packages as well

Comment (pratap0007) [reply ui/package.json:32]: I have updated only packages that has npm audit issue, if we look for other packages then there might be some breaking change in ui

Comment (PuneetPunamiya) [reply ui/package.json:32]: Yeah but I think with this patch we should also update the other packages if there is an updated version for it

Comment (PuneetPunamiya) [reply ui/package.json:32]: Also we can plan to bump the node and the npm version

**Comment by piyush-garg** [comment]
@pratap0007 Is this PR good for review or are you still working on it?

**Comment by pratap0007** [comment]
> @pratap0007 Is this PR good for review or are you still working on it?

Still working on it , issue is snapshot of Readme components changes some time and trying to fix it

File: ui/src/components/Readme/__snapshots__/Readme.test.tsx.snap (hunk 2 above)
Comment (pratap0007) [reply ui/src/components/Readme/__snapshots__/Readme.test.tsx.snap:26]: yes

**Comment by PuneetPunamiya** [comment]
Thanks 🤙🏻 
/approve
/meow

**Comment by vinamra28** [comment]
/hold
package-lock.json is changing for me when I am doing `npm install`. What's the npm and node version being used?

**Comment by pratap0007** [comment]
> /hold
> package-lock.json is changing for me when I am doing `npm install`. What's the npm and node version being used?

latest version for both (node  v16.8.0, npm  7.21.1)

**Comment by vinamra28** [comment]
/hold cancel
/lgtm
Thanks

**Pull request merged**